flatisfy/filters/__init__.py

```diff
@@ -67,6 +67,12 @@ def refine_with_housing_criteria(flats_list, constraint):
                             field.capitalize(), flat["id"])
             is_ok[i] = is_ok[i] and is_within_interval
 
+        # Check number of pictures
+        print(constraint['minimum_pictures'], len(flat['photos']))
+        if not (constraint['minimum_pictures'] <= len(flat['photos'])):
```

Comment:
With the change to support the `None` value, this can be written

```
if not tools.is_within_interval(flat.get('photos', []), constraint['minimum_pictures'], None):
```

I think calling `is_within_interval` makes it a bit more readable, and I'd rather use the `.get` form to ensure it does not fail if for some reason Weboob does not return a `photos` key.

Reply:
Actually, I looked at this change again, and I believe there is a bug here.

Flatisfy runs three passes:

1. First pass is to fetch infos from the listings (list of matching posts).
2. Second pass is to fetch infos from the details page (single post page).
3. Third pass is an advanced deduplication pass by checking multiple stuff, including photos. It does not matter for this PR.

`refine_with_housing_criteria` function is called at the end of both the first and the second passes. Problem is after the first pass there will likely be no photos fetched (or at most one, as there is at most one picture on the listing pages). Then, I fear that most of the posts will simply be discarded during the first pass, even if they should match and have a correct number of photos.

I think the best solution is to introduce a new function `refine_with_minimum_pictures_criterion` which would be called only at the end of the second pass.